src/formatDate.js

```diff
@@ -8,7 +8,28 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const partsOfDate = date.split(oldSeparator);
+  const dateObject = {};
+
+  for (let i = 0; i < 3; i++) {
+    dateObject[fromFormat[i]] = partsOfDate[i];
+  }
+
+  const newSeparator = toFormat[3];
+  const newPartsOfDate = toFormat.slice(0, 3).map((part) => {
```

Comment:
The mapping function is well-implemented, but it could be simplified by extracting the year conversion logic into a separate helper function. This would make the mapping logic more concise and easier to understand.